developer-docs-site/docs/sdks/transactions-with-ts-sdk.md

```diff
@@ -164,3 +142,31 @@ Alice coins: 5000. Should be 5000!
 Bob coins: 0. Should be 0!
 Bob coins: 717. Should be 717!
 ```
+
+## Build raw transactions with the ABI transaction builder
+
+To reduce the burden of serializing payload arguments manually, Typescript SDK also provides an ABI (Application Binary Interface) based transaction builder. ABI includes the information about the Move function signatures. With the help of ABI, Typescript SDK is able to serialize plain Javascript/Typescript values. ABI files are produced while compiling Move packages with the Aptos CLI. To build raw transactions with the ABI transaction builder, you need first need to convert the ABI files into hex strings. On linux/Mac, you can do it with below command:
+
+```bash
+cat aptos-core/aptos-move/framework/aptos-token/build/AptosToken/abis/token_transfers/offer_script.abi | od -v -t x1 -A n | tr -d ' \n'
+```
+
+And then, you can build raw transactions with:
+
+```ts
+import { TransactionBuilderABI, HexString } from "aptos";
+
+// You can pass in multiple ABIs
+const transactionBuilder = new TransactionBuilderABI([
+  new HexString("ABI_HEX_STRING_1").toUint8Array(),
+  new HexString("ABI_HEX_STRING_2").toUint8Array(),
+]);
```

Comment:
why aren't we pulling this from the chain like the rest endpoint does?

Reply:
No specific reasons why we cannot do that. But Ethereum's convention is to produce ABIs locally after compiling smart contracts. Tried to provide a similar experience.